http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -30,15 +30,13 @@ sealed trait ContainerRegistry extends Product with Serializable {
 }
 object ContainerRegistry {
   final case object GCR extends ContainerRegistry {
-    def regex: Regex = """(.*gcr.io)\/(.+)\/(.+)(\:?.*)$""".r
-
+    val regex: Regex = """^(.*gcr.io)/([\w.-]+/[\w.-]+)(?::(\w[\w.-]+))?(?:@([\w+.-]+:[A-Fa-f0-9]{32,}))?$""".r
     override def toString: String = "gcr"
   }
 
-// Repo format: https://docs.docker.com/docker-hub/repos/
-// Docker hub doesn't have documentation about length limitation on repo name and image name, but 100 max seems reasonable
+  // Repo format: https://docs.docker.com/docker-hub/repos/
   final case object DockerHub extends ContainerRegistry {
-    def regex: Regex = """^[A-Za-z0-9]+[\w./-]+(?::\w[\w.-]+)?(?:@[\w+.-]+:[A-Fa-f0-9]{32,})?$""".r
+    val regex: Regex = """^([\w.-]+/[\w.-]+)(?::(\w[\w.-]+))?(?:@([\w+.-]+:[A-Fa-f0-9]{32,}))?$""".r
```

Comment:
Added grouping and fixed some mistakes in the original regex: it would match things like `a////////`.

Reply:
Will do

Reply:
Added